### PR TITLE
fix(core): `Hint`/`Dropdown` fix directives order mattering

### LINKED ISSUE #2494: 🐞 - `Driver` order of attributes matter
### Which @taiga-ui/* package(s) are the source of the bug?

core

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/taiga-starter

### Is this issue blocking you?

Non-Blocking

### Description

We need to make `Driver` providers independent of the order of directives providing it.

### Angular version

any

### Taiga UI version

any

### Which browsers have you used?

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### LINKED ISSUE #3097: 🐞 - Conflict between `tuiDrodownContext` and `tuiHint` on same element
### Which @taiga-ui/* package(s) are the source of the bug?

core

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/angular-fim1dw?file=src%2Fapp%2Fapp.component.html

### Is this issue blocking you?

Non-Blocking

### Description

When using `tuiDrodownContext` and `tuiHint` on same element - there is no any (system or custom template via [tuDropdown]) context menu appears by right-clicking.
Removing `tuiHint` from element resolves that problem, but there is no hint in this way.

### Angular version

14.2.7

### Taiga UI version

3.10.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by waterplea**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #2494
Closes #3097

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by splincode** [comment]
- [ ] `pie-chart.cy.ts` 
- [ ] eslint `hint.providers.ts`

**Review by splincode — APPROVED** [review]

File: projects/core/abstract/abstract-driver.directive.ts [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
+import {merge, Observable} from 'rxjs';
 import {distinctUntilChanged, takeUntil} from 'rxjs/operators';
 
 import {TuiDriver} from './driver';
 import {TuiVehicle} from './vehicle';
 
 @Directive()
-export abstract class AbstractTuiDriverDirective {
+export abstract class AbstractTuiDriverDirective implements OnInit {
+    abstract type: string;
```
Comment (splincode) [inline projects/core/abstract/abstract-driver.directive.ts:11]: does it need to be unified?

```ts
abstract type: 'dropdown' | 'hint' | 'viewport';
```

Comment (waterplea) [reply projects/core/abstract/abstract-driver.directive.ts:11]: It's a general approach, can be anything really in future.

**Comment by splincode** [comment]
@waterplea 
image

**Review by vladimirpotekhin — APPROVED** [review]

**Comment by splincode** [comment]
image

**Pull request merged**